src/Model/Subscription.php

```diff
@@ -33,70 +36,67 @@ class Subscription extends Entity implements Arrayable
     public function __construct(?string $id = null)
     {
         parent::__construct($id);
+        $this->startDate = new DateTime('now', new DateTimeZone('UTC'));
+        $this->status = Status::initialized();
+        $this->customer = new Customer();
         $this->addOns = new Modifiers($this);
         $this->discounts = new Modifiers($this);
+        $this->plan = new NullPlan();
     }
 
     public function toArray(): array
     {
         $price = $this->getPrice();
-        $customer = $this->getCustomer();
         $payment = $this->getPaymentMethod();
-        $plan = $this->getPlan();
 
         return array_filter([
             'id' => $this->getId(),
             'startDate' => $this->getStartDate(),
             'price' => empty($price) ? null : $price->toArray(),
             'status' => $this->getStatus(),
-            'customer' => empty($customer) ? null : $customer->toArray(),
+            'customer' => $this->getCustomer()->toArray(),
             'payment' => empty($payment) ? null : $payment->toArray(),
-            'plan' => empty($plan) ? null : $plan->toArray(),
+            'plan' => $this->plan->toArray(),
             'addOns' => $this->addOns->toArray(),
             'discounts' => $this->discounts->toArray(),
         ]);
     }
 
     /**
      * Merge another Subscription into this one. Preserves the entity ID.
+     *
+     * @internal
      */
     public function merge(Subscription $other): Subscription
     {
         if ($startDate = $other->getStartDate()) {
             $this->setStartDate($startDate);
         }
 
-        if ($price = $other->getPrice()) {
+        $price = $other->getPrice();
+        if ($price && !$price instanceof NullPrice) {
             $this->setPrice($price);
         }
 
-        if ($status = $other->getStatus()) {
-            $this->setStatus($status);
-        }
+        $this->setStatus($other->getStatus());
 
-        if ($customer = $other->getCustomer()) {
-            $this->setCustomer($customer);
-        }
+        $this->setCustomer($other->getCustomer());
```

Comment:
should we allow changing customer on a subscription? not sure bt allows that?

Reply:
I marked `merge` as internal and I think I would like to axe it in a future release. It has become a bit of a confusing convenience method.

Reply:
It could be useful to merge "the same customer" in - especially if the "new" customer has more info than the previous.